src/comps/CompanyComparison/ComparisonPage.tsx

```
@@ -0,0 +1,136 @@
+import CombineGraphs, { GraphItem } from "./CombineGraphs";
+import GraphForm from "./GraphForm";
+import CompanyCheckboxList from "./CompanyCheckboxList";
+
+// import grid_texts from "../../texts/gridTexts.json";
+
+export enum graphsType {
+  bars,
+  scatter,
+  mixed,
+}
+
+export default function ComparisonPage() {
+  const [data, setData] = useState<Array<CompanySummery>>([]);
+  const [selectedCompanies, setSelectedCompanies] = useState(
+    [] as CompanySummery[]
+  );
+
+  const [graphs, setGraphs] = useState([] as Array<GraphItem>);
+
+  useEffect(() => {
+    getAllCompanies().then((companies) => {
+      setData(companies);
+      setSelectedCompanies(companies);
+    });
+  }, []);
+
+  const handleCompanyCheckBox = (checked: boolean, company: any) => {
+    checked
+      ? setSelectedCompanies([...selectedCompanies, company])
+      : setSelectedCompanies(
+          selectedCompanies.filter(
+            (selectedCompany) => selectedCompany.name !== company.name
+          )
+        );
+  };
+
+  const handleFormValues = (formValues: {
+    fieldOne: keyof CompanySummery;
+    fieldTwo: keyof CompanySummery;
+    graphCheckbox: graphsType;
+  }) => {
+    let graphData: GraphItem;
+    switch (formValues.graphCheckbox) {
+      case graphsType.bars:
+        graphData = {
+          type: graphsType.bars,
+          param: {
+            field: formValues.fieldOne,
+            title: formValues.fieldOne,
+            format: (val) => `${val} $`,
+          },
+        };
+        break;
+      case graphsType.mixed:
+        graphData = {
+          type: graphsType.mixed,
+          paramLeft: {
+            field: formValues.fieldOne,
+            title: formValues.fieldOne,
+            format: (val) => `${val} $`,
+          },
+          paramRight: {
+            field: formValues.fieldTwo,
+            title: formValues.fieldTwo,
+            format: (val) => `${val} FF`,
+          },
+        };
+        break;
+
+      default:
```

it is wrong to assume that this is the default :)
there are 3 options, no default, if you want/need the default it should throw an error